manifests/config.pp

```diff
@@ -21,13 +21,24 @@
     "main_collective" => $main_collective
   }
 
+  # some config settings are set directly on the main class, they are
+  # things other classes need to function for example libdir is also
+  # used by the plugin installer.  Specifically these settings should
+  # never be overridden by settings from the hashes since any difference
+  # between server/client cfg and what ends up being used by the plugin
+  # installer will lead to plugins not being found.
+  $global_config = {
```

Comment:
> yeah :( more piles of shit

This was why I did not want to deal with it in #111.

>  later more can be added

Right, I think lets stick to the absolute minimum as long as we cannot find a better way to deal with several configs.

But what about the collectives? Since client and server need them, we could at least put them here from the already existing parameters?